crates/iota-sdk-ffi/src/types/crypto/zklogin.rs

```diff
@@ -0,0 +1,334 @@
+    }
+
+    pub fn inputs(&self) -> ZkLoginInputs {
+        self.0.inputs.clone().into()
+    }
+
+    pub fn max_epoch(&self) -> u64 {
+        self.0.max_epoch
+    }
+
+    pub fn signature(&self) -> SimpleSignature {
+        self.0.signature.clone().into()
+    }
+}
+
+/// Public Key equivalent for Zklogin authenticators
+///
+/// A `ZkLoginPublicIdentifier` is the equivalent of a public key for other
+/// account authenticators, and contains the information required to derive the
+/// onchain account [`Address`] for a Zklogin authenticator.
+///
+/// ## Note
+///
+/// Due to a historical bug that was introduced in the IOTA Typescript SDK when
+/// the zklogin authenticator was first introduced, there are now possibly two
+/// "valid" addresses for each zklogin authenticator depending on the
+/// bit-pattern of the `address_seed` value.
+///
+/// The original bug incorrectly derived a zklogin's address by stripping any
+/// leading zero-bytes that could have been present in the 32-byte length
+/// `address_seed` value prior to hashing, leading to a different derived
+/// address. This incorrectly derived address was presented to users of various
+/// wallets, leading them to sending funds to these addresses that they couldn't
+/// access. Instead of letting these users lose any assets that were sent to
+/// these addresses, the IOTA network decided to change the protocol to allow
+/// for a zklogin authenticator who's `address_seed` value had leading
+/// zero-bytes be authorized to sign for both the addresses derived from both
+/// the unpadded and padded `address_seed` value.
+///
+/// # BCS
+///
+/// The BCS serialized form for this type is defined by the following ABNF:
+///
+/// ```text
+/// zklogin-public-identifier-bcs = bytes ; where the contents are defined by
+///                                       ; <zklogin-public-identifier>
+///
+/// zklogin-public-identifier = zklogin-public-identifier-iss
+///                             address-seed
+///
+/// zklogin-public-identifier-unpadded = zklogin-public-identifier-iss
+///                                      address-seed-unpadded
+///
+/// ; The iss, or issuer, is a utf8 string that is less than 255 bytes long
+/// ; and is serialized with the iss's length in bytes as a u8 followed by
+/// ; the bytes of the iss
+/// zklogin-public-identifier-iss = u8 *255(OCTET)
+///
+/// ; A Bn254FieldElement serialized as a 32-byte big-endian value
+/// address-seed = 32(OCTET)
+///
+/// ; A Bn254FieldElement serialized as a 32-byte big-endian value
+/// ; with any leading zero bytes stripped
+/// address-seed-unpadded = %x00 / %x01-ff *31(OCTET)
+/// ```
+///
+/// [`Address`]: crate::Address
+#[derive(Clone, Debug, derive_more::From, uniffi::Object)]
+pub struct ZkLoginPublicIdentifier(pub iota_types::ZkLoginPublicIdentifier);
+
+#[uniffi::export]
+impl ZkLoginPublicIdentifier {
+    #[uniffi::constructor]
+    pub fn new(iss: String, address_seed: &Bn254FieldElement) -> Result<Arc<Self>> {
```

Comment:
Do we really need the Arc here?

Reply:
Why? How is that different from `ZkLoginInputs::new` that don't need it? Also, I removed it locally and it's compiling, what am I missing?